esp32-hal/examples/aes.rs

```diff
@@ -35,19 +35,17 @@ fn main() -> ! {
     let mut block_buf = [0_u8; 16];
     block_buf[..plaintext.len()].copy_from_slice(plaintext);
 
-    let key = Key::<Aes128>::from(&keybuf);
-    let mut cipher = Cipher::new(&mut aes, &key);
     let mut block = block_buf.clone();
     let pre_hw_encrypt = xtensa_lx::timer::get_cycle_count();
-    cipher.encrypt_block(&mut block);
+    aes.start_operation(&mut block, Mode::Encryption128, &keybuf);
```

Comment:
`start_operation` is a bit misleading for a blocking operation that is finished when it returns
`start_xxx` would imply there is a way to finish the operation later